ui/app/js/controllers/user_edit_controller.js

```diff
@@ -100,11 +100,14 @@ function ($scope, $modalInstance, CSRF, Permissions, user, users) {
           role: '',
           data_version: ''
         };
+        $scope.errors = {
+          roles: {}
+        };
         sweetAlert("Saved", "Role granted.", "success");
       })
       .error(function(response) {
         if (typeof response === 'object') {
-          $scope.errors = response;
+          $scope.errors.role = response;
```

Comment:
I see you've added a "roles" object to $scope.errors, above, but you're reference "role" here, that probably isn't helping.